onecodex/models/collection.py

```diff
@@ -274,6 +289,25 @@ def _collate_results(self, field=None, include_host=None):
             table = results["table"]
             host_tax_ids = results.get("host_tax_ids", [])
 
+            table = {t.get("tax_id"): t for t in table}
```

Comment:
Can there be cases where `t` doesn't have tax_id? Just a bit below it's doing `["tax_id"]`